### DISCUSSION TITLE
Can a watermarked file be restored to its original, watermark-free state

### DISCUSSION THREAD

**Discussion by YoonAHMM** [swesterfeld/audiowmark#63] [Q&A]
"Is the modification made by audiowmark irreversible? Can it be restored to its original state through any operation?"

**Comment by tim-janik** [comment]
Currently, no such functionality is implemented.

While some of the artefacts introduced by the watermarking process could be reduced, given the key, watermark and exact WAV output of audiowmark is known, components such as the limiter employed at the final stage are not exactly reversible. The stages are documented here:
    https://github.com/swesterfeld/audiowmark/blob/master/docs/audiowmark.md

Generally, someone who possesses the key and the watermark applied to an audiofile also has the original audiofile itself.
So perfectly reversing a watermark usually isn't within the scope of the design of watermarking schemes.
